src/main/scala/com/nawforce/apexlink/org/PackageAPI.scala

```diff
@@ -168,10 +168,13 @@ trait PackageAPI extends Package {
   }
 
   /** Get a array of type identifiers from this packages modules. */
-  override def getTypeIdentifiers: Array[TypeIdentifier] = {
+  override def getTypeIdentifiers(apexOnly: Boolean): Array[TypeIdentifier] = {
     modules
-      .foldLeft(Set[TypeIdentifier]())((acc, module) => acc ++ module.getMetadataDefinedTypeIdentifiers)
-      .toArray
+      .foldLeft(Set[TypeIdentifier]())((acc, module) =>
+        acc ++ module.getMetadataDefinedTypeIdentifiers.filter(id =>
+          !apexOnly || TypeResolver(id.typeName, module).toOption.exists(
```

Comment:
Think would be better to push this flag into getMetadataDefinedTypeIdentifiers as that would be more efficient that using the TypeResolver. There is a collect in getMetadataDefinedTypeIdentifiers that you can just add a condition to avoid picking up SObjects.

Reply:
Moved the test of apex only to getMetadataDefinedTypeIdentifiers